### PR TITLE
Adding Vector Type

### PR DESCRIPTION
**Pull request by komalon**

### CODE REVIEW

**Comment by SabirAkhadov** [comment]
Is this getting merged soon? We could make use of the vector type for implementing correct struct ABI for Cfunc.

**Comment by seibert** [comment]
@sklam: We also have the much older PR #313.  Not sure if some of that should be pulled forward into this PR.

File: llvmlite/ir/types.py [hunk 1]
```diff
@@ -196,6 +196,39 @@ def __eq__(self, other):
     def __hash__(self):
         return hash(FunctionType)
 
+class VectorType(Type):
+
+    @classmethod
+    def __new__(cls, count, typ):
```
Comment (sklam) [inline llvmlite/ir/types.py]: There's no need to use `__new__` here.  This should be a `__init__` instead.    Also, a `__new__` doesn't need `@classmethod`.

File: llvmlite/ir/types.py [hunk 2]
```diff
@@ -196,6 +196,39 @@ def __eq__(self, other):
...
+        assert isinstance(count, int) and count >= 0
+        assert isinstance(typ, Type)
+        self = super(VectorType, cls).__new__(cls)
+        self.count = count
+        self.type = typ
+        return self
+
+    def __getnewargs__(self):
```
Comment (sklam) [inline llvmlite/ir/types.py]: Don't need this if `__new__` is not defined.

**Pull request merged**